repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/85sys-upgrade-redhat/mount_usr.sh

```diff
@@ -38,8 +36,14 @@ mount_usr()
                     _dev="/dev/disk/by-uuid/${_dev#UUID=}"
                     ;;
             esac
+
+            # TODO: -ef replaced with inode number comparsion - is this enough?
+            # TODO: supresses all shellcheck warnings regarding declared, but
+            # not assigned variables in the whole if statement - is this the
+            # right way of dealing with this warning?
+            # shellcheck disable=SC2154 # Variable root is assigned by dracut
             if strstr "$_opts" "subvol=" && \
-                [ "${root#block:}" -ef $_dev ] && \
+                [ "`stat -c '%i' ${root#block:}`" = "`stat -c '%i' $_dev`" ] && \
```

Comment:
The -ef isn't defined in POSIX, so this should be equivalent way of checking, however might be false positive if the two files are on different filesystems but happen to have the same inode.

Reply:
If you have a look at what is the `read` reading you see that is reading some `/etc/fstab`. Therefore the `$_dev` variable should point to a device inside the `/dev` (devtmpfs filesystem). Similarly, `root` variable contains the name of partition containing the root filesystem, and therefore, should be also in `/dev`. (IIUC the `root` variable should contain the value of `root` cmdline argument). Therefore, both of these variable should point somewhere inside the same filesystem (`/dev`), meaning that inode number collisions are highly unlikely.